src/dbnode/encoding/scheme.go

```diff
@@ -44,7 +44,7 @@ var (
 
 	// TODO(xichen): set more reasonable defaults once we have more knowledge
 	// of the use cases for time units other than seconds.
-	defaultTimeEncodingSchemes = map[xtime.Unit]TimeEncodingScheme{
+	defaultTimeEncodingSchemes = []TimeEncodingScheme{
```

Comment:
Hey so this part was super hacky when we threw it together, may need to make this more resilient... what would you think of keeping this as a map like before, then add a function that would transform 

```golang
func NewTimeEncodingSchemes(schemes map[xtime.Unit]TimeEncodingScheme) TimeEncodingSchemes {
 schemes := make(TimeEncodingSchemes, int(xtime.Year))
 for k, v := range schemes {
  if k <= xtime.None || k > xtime.Year {
   schemes[k] = v
  }

 return schemes
}
```